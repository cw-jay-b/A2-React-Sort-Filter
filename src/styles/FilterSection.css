.filter-group {
  background-color: #f9f9f9;
  padding: 1rem;
  display: flex;
  flex-direction: column;
  row-gap: 0.5rem;
  margin-bottom: 1rem;
}

.filter-group label {
  display: block;
  margin-bottom: 5px;

  &:hover{
    cursor: pointer;
  }
}

input[type="number"] {
  width: 4rem;
  margin: 0 5px;
  padding: 5px;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.min-max-box, .sort-input-box{
  display: flex;
  background-color: #fff;
  border: 1px solid rgb(108, 108, 108);
  align-items: center;
  border-radius: 0.25rem ;
  padding: 0.5rem;
}

.sort-wrapper {
  border: 1px solid lightgray;
  border-radius: 0.5rem;
  background-color: #f9f9f9;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  column-gap: 0.5rem;
  padding: 0.5rem 1rem;
}
.sort-wrapper select:hover{
  cursor: pointer;
}
#min-box{
  margin-right: 1rem;
}
#max-box{
  margin-left: 1rem;
}

#min-price-input, #max-price-input {
  /* margin: 0.5rem 0.5rem; */
  width: 2.5rem;
  background-color: transparent;
  outline: none;
  border: none;
}
