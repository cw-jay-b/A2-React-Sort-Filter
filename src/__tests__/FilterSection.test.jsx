import React from 'react';
import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';

import FilterSection from '../components/FilterSection';

describe('Sort Filter Tests', ()=>{
it('should match snapshot with default props', () => {
    const { asFragment } = render(
        <FilterSection
        selectedFuel={[]}
        setSelectedFuel={() => {}}
        setPriceRange={() => {}}
        handleClearAll={() => {}}
        resultCount={0}
        />
    );
    expect(asFragment()).toMatchSnapshot();
    });
    
    it('should match snapshot with selected fuels and price range', () => {
    const { asFragment } = render(
        <FilterSection
        selectedFuel={[1, 3]} // Example of selected fuels
        setSelectedFuel={() => {}}
        setPriceRange={() => {}}
        handleClearAll={() => {}}
        resultCount={10}
        />
    );
    expect(asFragment()).toMatchSnapshot();
    });
});