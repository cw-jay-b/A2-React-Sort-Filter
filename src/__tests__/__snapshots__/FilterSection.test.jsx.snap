// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sort Filter Tests > should match snapshot with default props 1`] = `
<DocumentFragment>
  <aside
    class="filter-section"
    style="position: sticky; top: 1rem;"
  >
    <div
      style="margin-bottom: 1rem; display: flex; justify-content: space-between;"
    >
      <span>
        <img
          alt="filter_icon"
          src="/src/assets/icons8-filter-16.png"
          style="margin-right: 0.5rem;"
        />
        <span
          style="font-size: 1.25rem; font-weight: bold;"
        >
          Filters
        </span>
      </span>
      <button
        style="color: rgb(9, 121, 182); background: none; outline: none; cursor: pointer;"
      >
        Clear All
      </button>
    </div>
    <div
      class="filter-group"
    >
      <h4>
        Fuel
      </h4>
      <label>
        <input
          style="margin: 0px 0.5rem;"
          type="checkbox"
          value="1"
        />
        Petrol
      </label>
      <label>
        <input
          style="margin: 0px 0.5rem;"
          type="checkbox"
          value="2"
        />
        Diesel
      </label>
      <label>
        <input
          style="margin: 0px 0.5rem;"
          type="checkbox"
          value="3"
        />
        CNG
      </label>
      <label>
        <input
          style="margin: 0px 0.5rem;"
          type="checkbox"
          value="4"
        />
        LPG
      </label>
      <label>
        <input
          style="margin: 0px 0.5rem;"
          type="checkbox"
          value="5"
        />
        Electric
      </label>
      <label>
        <input
          style="margin: 0px 0.5rem;"
          type="checkbox"
          value="6"
        />
        Hybrid
      </label>
    </div>
    <div
      class="filter-group"
      id="min-max-filter"
      style="margin-bottom: 0px;"
    >
      <h4>
        Price Range
      </h4>
      <div
        style="display: flex; align-items: center;"
      >
        <span
          class="min-max-box"
          id="min-box"
        >
          <input
            id="min-price-input"
            min="0"
            name="min"
            placeholder="1"
            type="number"
          />
          Lakhs
        </span>
         
        <span>
          -
        </span>
        <span
          class="min-max-box"
          id="max-box"
        >
          <input
            id="max-price-input"
            min="0"
            name="max"
            placeholder="5"
            type="number"
          />
          Lakhs
        </span>
      </div>
    </div>
    <div
      class="results-count"
    >
      Showing 0 Car 
    </div>
  </aside>
</DocumentFragment>
`;

exports[`Sort Filter Tests > should match snapshot with selected fuels and price range 1`] = `
<DocumentFragment>
  <aside
    class="filter-section"
    style="position: sticky; top: 1rem;"
  >
    <div
      style="margin-bottom: 1rem; display: flex; justify-content: space-between;"
    >
      <span>
        <img
          alt="filter_icon"
          src="/src/assets/icons8-filter-16.png"
          style="margin-right: 0.5rem;"
        />
        <span
          style="font-size: 1.25rem; font-weight: bold;"
        >
          Filters
        </span>
      </span>
      <button
        style="color: rgb(9, 121, 182); background: none; outline: none; cursor: pointer;"
      >
        Clear All
      </button>
    </div>
    <div
      class="filter-group"
    >
      <h4>
        Fuel
      </h4>
      <label>
        <input
          checked=""
          style="margin: 0px 0.5rem;"
          type="checkbox"
          value="1"
        />
        Petrol
      </label>
      <label>
        <input
          style="margin: 0px 0.5rem;"
          type="checkbox"
          value="2"
        />
        Diesel
      </label>
      <label>
        <input
          checked=""
          style="margin: 0px 0.5rem;"
          type="checkbox"
          value="3"
        />
        CNG
      </label>
      <label>
        <input
          style="margin: 0px 0.5rem;"
          type="checkbox"
          value="4"
        />
        LPG
      </label>
      <label>
        <input
          style="margin: 0px 0.5rem;"
          type="checkbox"
          value="5"
        />
        Electric
      </label>
      <label>
        <input
          style="margin: 0px 0.5rem;"
          type="checkbox"
          value="6"
        />
        Hybrid
      </label>
    </div>
    <div
      class="filter-group"
      id="min-max-filter"
      style="margin-bottom: 0px;"
    >
      <h4>
        Price Range
      </h4>
      <div
        style="display: flex; align-items: center;"
      >
        <span
          class="min-max-box"
          id="min-box"
        >
          <input
            id="min-price-input"
            min="0"
            name="min"
            placeholder="1"
            type="number"
          />
          Lakhs
        </span>
         
        <span>
          -
        </span>
        <span
          class="min-max-box"
          id="max-box"
        >
          <input
            id="max-price-input"
            min="0"
            name="max"
            placeholder="5"
            type="number"
          />
          Lakhs
        </span>
      </div>
    </div>
    <div
      class="results-count"
    >
      Showing 10 Cars 
    </div>
  </aside>
</DocumentFragment>
`;
